tests/apps/core/test_core_api_room_user_access.py

```diff
@@ -163,171 +358,504 @@ def test_api_room_user_accesses_create_anonymous(self):
             },
             HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
         )
 
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(
+            response.json(),
+            {
+                "detail": "You must be administrator or owner of a room to add accesses to it."
+            },
+        )
+        self.assertFalse(RoomUserAccess.objects.filter(user=other_user).exists())
+
+    def test_api_room_user_accesses_create_administrators_except_owner_direct(self):
+        """
+        A user who is direct administrator in a room should be allowed to create
+        room user accesses in this room for roles other than owner.
+        """
+        user, other_user = UserFactory.create_batch(2)
+        room = RoomFactory(users=[(user, "administrator")])
+
+        jwt_token = AccessToken.for_user(user)
+
+        response = self.client.post(
+            "/api/room-user-accesses/",
+            {
+                "user": str(other_user.id),
+                "room": str(room.id),
+                "role": random.choice(["member", "administrator"]),
+            },
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
         self.assertEqual(response.status_code, 201)
         self.assertEqual(RoomUserAccess.objects.count(), 2)
         self.assertTrue(RoomUserAccess.objects.filter(user=other_user).exists())
 
-    def test_api_room_user_accesses_create_administrator_groups(self):
+    def test_api_room_user_accesses_create_administrators_except_owner_via_group(self):
         """
-        Users who are administrators of a room via a group should be allowed to create a room
-        user access in it."""
-        user = UserFactory()
+        A user who is administrator in a room via a group should be allowed to create
+        room user accesses in this room for roles other than owner.
+        """
+        user, other_user = UserFactory.create_batch(2)
         group = GroupFactory(members=[user])
-        room = RoomFactory(groups=[(group, True)])
-        other_user = UserFactory()
-        is_administrator = random.choice([True, False])
+        room = RoomFactory(groups=[(group, "administrator")])
 
         jwt_token = AccessToken.for_user(user)
 
         self.assertFalse(RoomUserAccess.objects.exists())
         response = self.client.post(
-            f"/api/rooms/{room.id!s}/users/",
+            "/api/room-user-accesses/",
             {
                 "user": str(other_user.id),
-                "is_administrator": is_administrator,
+                "room": str(room.id),
+                "role": random.choice(["member", "administrator"]),
             },
             HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
         )
         self.assertEqual(response.status_code, 201)
         self.assertEqual(RoomUserAccess.objects.count(), 1)
         self.assertTrue(RoomUserAccess.objects.filter(user=other_user).exists())
 
```

Comment:
ditto for owner

Reply:
I didn't get this comment.

Reply:
I added an explanation to the docstring that the owner role is tested in the subsequent test.